src/tests/unit/test_reference.py

```diff
@@ -42,90 +44,54 @@ def mock_update(entity: dict, *args, **kwargs):
             "type": SIMOS.REFERENCE.value,
             "referenceType": REFERENCE_TYPES.LINK.value,
         }
-        document_service.insert_reference(
-            Address("$1.uncontained_in_every_way", "testing"),
-            ReferenceEntity.parse_obj(reference),
+        document_service.update_document(
+            Address("$1.uncontained_in_every_way", "testing"), reference, update_uncontained=False
         )
         assert doc_storage["1"]["uncontained_in_every_way"] == reference
 
-    def test_insert_reference_target_does_not_exist(self):
```

Comment:
document update does not have validation for checking if reference target exists